src/main/scala/com/daml/projection/javadsl/BatchSource.scala

```diff
@@ -162,12 +173,23 @@ object BatchSource {
       (exercisedEvent: J.ExercisedEvent) => Optional.of(exercisedEvent.getTemplateId)
   }
 
+  /**
+   * Returns an parties from `E`.
+   *
+   * It should be only used by [[BatchSource]] create methods which creates [[BatchSource]] for testing purposes. It can
```

Comment:
It is used in [[BatchSource]] `create` methods used for testing purposes, where the [[BatchSource]] is created from in-memory data structures.